### PR TITLE
Virtual schema docs

### LINKED ISSUE #2957: Document how to use crdb_internal
In trying to figure out how to read from the `information_schema` table, it was suggested by @benesch that it might be beneficial to teach people how to use `crdb_internal`.

Taken from slack:

```
My opinion is that it probably makes more sense to teach people to use crdb_internal
information_schema is technically "stable" but it has a bunch of random crap because it needs to describe any SQL database ever
And crdb_internal only has stuff that's actually relevant to Cockroach
```

### LINKED ISSUE #8706: Document virtual schemas
We need more documentation on virtual schemas, especially since we use them in a lot of examples and direct guidance. At the very least, we should have a list of the virtual schemas in one place, with a brief description of each, and a link to more detailed docs, if available (perhaps a subsection of https://www.cockroachlabs.com/docs/stable/sql-name-resolution.html).

### LINKED ISSUE #9569: sql: format arrays in pg_catalog to match Postgres
PR: https://github.com/cockroachdb/cockroach/pull/56980

From release notes:
> Arrays in `pg_catalog` tables now are diaplayed in a format that matches PostgreSQL. [#56980][#56980] {% comment %}doc{% endcomment %}

Related to https://cockroachlabs.atlassian.net/browse/CRDB-1523.

### LINKED ISSUE #9578: sql: populated pg_depend with table and view dependencies
PR: https://github.com/cockroachdb/cockroach/pull/57240

From release notes:
> Added table-view dependency information in `pg_depend` to improve compatibility with PostgreSQL. [#57240][#57240] {% comment %}doc{% endcomment %}

Related to https://cockroachlabs.atlassian.net/browse/CRDB-1523.

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #2957.
Fixes #8706.
Fixes #9569. 
Fixes #9578.

Update of https://github.com/cockroachdb/docs/pull/5454.

### CODE REVIEW

**Comment by ericharmeling** [comment]
@glennfawcett FYI

**Review by knz — COMMENTED** [review]
Before I go into the details of the contents, there are three high-level concerns here:

1. I am surprised that `pg_catalog` does not get documented. We are committed to supporting `pg_catalog`, and in fact it gives access to *more* useful data than `info_schema` / `crdb_internal`. Moreover, it happens to help more with applications already developed for postgres. What is the plan about that?

2. there's been recent development, which I hope @piyush-singh can help clarify: that we do not wish to promote `crdb_internal` as a supported SQL API for use by end-users. This means that documentation, if any (we can discuss if it should be present at all) should disclaim loudly that all this information does not come with any guarantee of stability or preservation across versions.  Hence the following questions:

   - is it desirable to have `crdb_internal` documented here at all?
   - if yes, what is the best way to properly communicate that this is provided with many strings attached?

3. each of the `crdb_internal` vtables come with a CPU, memory and latency cost associated, which in some cases can impact cluster stability or performance upon use. This must absolutely be called out if we decide to document this virtual schema. (the cost functions are displayed in the table comments)

> Reviewed 10 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8909#-:-MM00XwN6_OYorc3m07s:b-5d948d)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling)

**Review by ericharmeling — COMMENTED** [review]
Thanks for bringing these concerns forward, @knz !

Some quick responses to your points:

1. I am more than happy to add docs for `pg_catalog` to this PR. 

2. I think product should make the call on whether or not we want to encourage our end users to use `crdb_internal`. If we want to discourage/not encourage its use, then we definitely shouldn't document the schemas contents in detail.

3. Okay! If we do choose to document the schema, I can port all of the information from those table comments (how fortuitous) into the document.

Some additional thoughts:

Throughout our docs, we use `crdb_internal` tables in examples that require certain information about a cluster that is not exposed by SQL statements or other virtual schema tables. There might have been some new statements introduced over the past few releases that make these `crdb_internal` tables less useful. 

If we want to *not* document `crdb_internal`, then I can open an issue to remove (or replace, where possible) references to `crdb_internal` tables. We can then limit its documentation to brief mentions in the existing [Name Resolution](https://www.cockroachlabs.com/docs/stable/sql-name-resolution.html) page (and possibly the proposed Virtual Schemas overview page).

If we do want to document `crdb_internal`, then I can modify the documentation introduced in this PR to have no stability distinctions among tables in `crdb_internal`. I can also add a warning to the top of the page calling out the lack of stability guarantees. And also additional documentation (a column in the table of tables, and possibly a short paragraph) concerning the costs associated with using `crdb_internal` tables.

I'll wait on the final word from @piyush-singh before moving forward on this.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8909#-:-MMMum9U6oD1l6IR-K_T:b-yj27nj)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Comment by ericharmeling** [comment]
@piyush-singh Ping on whether or not we want to document `crdb_internal`.

**Comment by ericharmeling** [comment]
@knz 
- I added a page on `pg_catalog`. I'm not sure how much more information we want to provide on that schema.
- I updated the page on `crdb_internal` to include a warning re: stability and performance, and less detailed information about the tables in the schema.

@piyush-singh 
Ping on the questions in the comment above re: `crdb_internal` docs.

**Review by knz — COMMENTED** [review]
> Throughout our docs, we use crdb_internal tables in examples that require certain information about a cluster that is not exposed by SQL statements or other virtual schema tables. 

This is a strong signal that something is amiss in our product management. If there was a user need to see some information exposed, the proper "next action" was to ensure we created some SHOW statement or did some research to see if `pg_catalog` could be used. Adding documentation on how to use `crdb_internal` was not the right approach.

I would recommend some investigation by PM about this. @piyush-singh what do you think?

> There might have been some new statements introduced over the past few releases that make these crdb_internal tables less useful.

Yes that might be the case too. But then if you have not been informed of this, that is perhaps another defect in our process as we should have had github/jira tickets to address these missing features and you would have been informed via release notes.

> I added a page on pg_catalog. I'm not sure how much more information we want to provide on that schema.

I'd say we want to document everything we support, and highlight in particular those columns which exist in our schema but which we have not yet populated (i.e. they contain NULL values), because they may be sources of incompatibilities with postgres.

We also want to remind users to talk to us if they find they would like to use pg_catalog but some aspect of it is still missing.

> Reviewed 1 of 3 files at r2, 8 of 8 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8909#-:-MT_rPXbAQOlEto_n9sZ:beejgl2)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling)

___
*[v20.1/crdb-internal.md, line 15 at r3](https://reviewable.io/reviews/cockroachdb/docs/8909#-MT_sAF91S5nTsa_N9dl:-MT_sAF91S5nTsa_N9dm:bky3cuu) ([raw file](https://github.com/cockroachdb/docs/blob/bb64665a60832d743e2963383b9e85d97e87b4b9/v20.1/crdb-internal.md#L15)):*
> ```Markdown
> {{site.data.alerts.callout_danger}}
> We do not recommend using `crdb_internal` tables in production environments for the following reasons:
> - The contents of `crdb_internal` schema are unstable, and subject to change in new releases of CockroachDB.
> ```

to change without prior notice
___
*[v20.1/crdb-internal.md, line 16 at r3](https://reviewable.io/reviews/cockroachdb/docs/8909#-MT_sFS4Ai3ccDDWkaBE:-MT_sFS51jqwC2cGxChi:b-800qy4) ([raw file](https://github.com/cockroachdb/docs/blob/bb64665a60832d743e2963383b9e85d97e87b4b9/v20.1/crdb-internal.md#L16)):*
> ```Markdown
> We do not recommend using `crdb_internal` tables in production environments for the following reasons:
> - The contents of `crdb_internal` schema are unstable, and subject to change in new releases of CockroachDB.
> - There are memory and latency costs associated with each table in `crdb_internal`. Accessing the tables in the schema can impact cluster stability and performance.
> ```

This particular point is also generally true of SHOW statements and other features. I would rather see callouts about memory usage / performance for each individual feature that incurs them.

**Comment by jseldess** [comment]
@ericharmeling, what's the current state of this PR?

**Comment by ericharmeling** [comment]
@jseldess 
> what's the current state of this PR?

TL;DR:

It has been on hold, pending further direction/prioritization from product management. I just synced with @vy-ton and now have a clearer direction for this work, as a result of [this 21.1 product roadmap item](https://cockroachlabs.atlassian.net/browse/CRDB-1523).

More detailed response:

I originally started working on virtual schema docs in 2019, in response to several requests for `crdb_internal` documentation. That work was not a priority, so I closed my first PR on virtual schema docs (https://github.com/cockroachdb/docs/pull/5454). I then received several more requests for `crdb_internal` docs, and so I opened a new PR for virtual schema docs (i.e., this PR).

Despite the longstanding requests for `crdb_internal` docs, whether we want to document `crdb_internal` at all is still in question. @knz gave some legitimate reasons why we would not want to expose `crdb_internal` virtual schema in docs. In summary: `crdb_internal` is an interface to internals of the product, its tables are likely to change across releases, and there are better virtual schemas to which to point users (e.g., `pg_catalog`). So I am thinking that we *do not* want to document `crdb_internal` in detail. 

On the topic of virtual schemas, we definitely *do* want to document `pg_catalog`. It seems to be the most important virtual schema for end users, as it provides useful information (as noted above by @knz), its tables correspond to the [system catalogs in PostgreSQL](https://www.postgresql.org/docs/10/catalogs-overview.html), and improvements to the table are part of the 21.1 product roadmap (see https://cockroachlabs.atlassian.net/browse/CRDB-1523). This PR includes a page on `pg_catalog`, but it could use more detail. Note that in 21.1, we are adding tables to the virtual schema that will be empty. Users should know that these empty tables are included for PG compatibility, but remain unimplemented.

So the plan moving forward is:

1. Remove the `crdb_internal` page from this PR.
2. Add more detail to the `pg_catalog` page about exactly which tables we support, and which ones we do not support.
3. Add a note to the 21.1 `pg_catalog` page about the empty tables.

**Comment by jseldess** [comment]
Thanks for this update, @ericharmeling!

**Comment by ericharmeling** [comment]
Updates to `v21.1/pg-catalog.md` are pending the finalization of the new system catalog tables added to `pg_catalog` in 21.1. 

See here for a tracking doc (still working on this doc, so let me know if you want to add anything/change anything before you do): https://docs.google.com/spreadsheets/d/1azfgQJQJriUii1dd8_RwhgHBqhlrwkjao1rISR8YAOc/edit#gid=1321389238

**Comment by ericharmeling** [comment]
There will also be subsequent PRs to update `v21.1/information-schema.md`. I'm probably going to wait until all work on `information_schema` has been finalized before starting on that.

**Comment by ericharmeling** [comment]
See https://github.com/cockroachdb/cockroach/issues/61750.

**Comment by ericharmeling** [comment]
@RichardJCai 

Adding you to review `v21.1/pg-catalog.md`.

**Comment by ericharmeling** [comment]
@jordanlewis @vy-ton 

Adding both of you for a general review, following our conversation today (3/15/21).

**Review by vy-ton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8909)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @jordanlewis, @knz, and @RichardJCai)

___
*[_includes/sidebar-data-v20.2.json, line 2422 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw0WyQ-NxMrF7reNtj:-MVw0WyQ-NxMrF7reNtk:bbjtsvt) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/_includes/sidebar-data-v20.2.json#L2422)):*
> ```JSON
>               },
>               {
>                 "title": "Virtual Schemas",
> ```

2 suggestions:
* Put this closer to `Name Resolution` as it's the closest relative
* We should get Engineering input but I've been referring to this as `System Catalog`. Developers might not understand "virtual".
___
*[v20.2/information-schema.md, line 9 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw21atAW-wbphfdub2:-MVw21atAW-wbphfdub3:bz04xei) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v20.2/information-schema.md#L9)):*
> can therefore be relied on to remain stable over time

We're not quite here yet with `information_schema`. I'm looking into this so will have more details soon
___
*[v20.2/pg-catalog.md, line 7 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw2pCi1UJjnKKxDe-Q:-MVw2pCi1UJjnKKxDe-R:ba7y209) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v20.2/pg-catalog.md#L7)):*
> ```Markdown
> ---
> 
> For PostgreSQL compatibility, CockroachDB includes a [virtual schema](virtual-schemas.html) called `pg_catalog`. The tables in the `pg_catalog` schema roughly correspond to the [system catalogs in PostgreSQL](https://www.postgresql.org/docs/13/catalogs-overview.html). `pg_catalog` tables are read-only.
> ```

They correspond to both system catalog and system views, maybe use https://www.postgresql.org/docs/13/catalogs.html as the link
___
*[v20.2/pg-catalog.md, line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw3CHW-MuD3kWq10Oq:-MVw3CHW-MuD3kWq10Or:bff7223) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v20.2/pg-catalog.md#L13)):*
> ```Markdown
> The tables in `pg_catalog` correspond to a subset of the virtual tables and views that make up the PostgreSQL system catalogs. Not all PostgreSQL system catalogs have a corresponding table in `pg_catalog`. See the table below for a detailed comparison between `pg_catalog` tables and the PostgreSQL 13 system catalogs.
> 
> PostgreSQL 13 system catalog | `pg_catalog` table  
> ```

Looking into the `None` tables [here](https://docs.google.com/document/d/1Y_S6-S_QfNewu70pil42XYU7rWLOwO0HAFiqtAL6Iao/edit#heading=h.dt3nanno7v0e), since I expected to be able to query empty `pg_catalog` tables
___
*[v21.1/information-schema.md, line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw4ibcAWw2ctxh6u3i:-MVw4ibcAWw2ctxh6u3j:b-rvi3f3) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v21.1/information-schema.md#L13)):*
> ```Markdown
> ## Data exposed by `information_schema`
> 
> To perform introspection on objects, you can either read from the related `information_schema` table or use one of CockroachDB's `SHOW` statements. `information_schema` tables are read-only.
> ```

@richardjcai Is `information_schema` what powers `SHOW` commands?
___
*[v21.1/pg-catalog.md, line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw5Bxb21lr4un1wah4:-MVw5Bxb21lr4un1wah5:b307tvk) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v21.1/pg-catalog.md#L13)):*
> ```Markdown
> The tables in `pg_catalog` correspond to a subset of the virtual tables and views that make up the PostgreSQL system catalogs. Not all PostgreSQL system catalogs have a corresponding table in `pg_catalog`, and not all `pg_catalog` tables have been fully implemented. See the table below for a detailed comparison between `pg_catalog` tables and the PostgreSQL 13 system catalogs.
> 
> PostgreSQL 13 system catalog | `pg_catalog` table  
> ```

How are you determining the list for the first column? We might be falling into a trap of duplicating the list of Postgres pg_catalog. Instead of just documenting the ones we do implement.

File: v20.2/pg-catalog.md
Comment (vy-ton) [inline v20.2/pg-catalog.md:13]: We should clarify what `None` here means. We have added tables like `pg_amop` as empty tables. So they are unimplemented but they do output in `SHOW TABLES` and you can query the empty table

**Comment by RichardJCai** [comment]
> @RichardJCai Is `information_schema` what powers `SHOW` commands?

`crdb_internal.create_statements` powers the `SHOW` commands for tables.

File: v21.1/pg-catalog.md
Comment (RichardJCai) [inline v21.1/pg-catalog.md:17]: LGTM but piggybacking on Vy's comment, I think we should clarify that unimplemented means that the table exists but the rows are not populated.

**Review by vy-ton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8909)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @jordanlewis, and @knz)

___
*[v21.1/information-schema.md, line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw4ibcAWw2ctxh6u3i:-MVxiF9y5uYiRaVdo3kA:b8pdi4i) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v21.1/information-schema.md#L13)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

@richardjcai Is `information_schema` what powers `SHOW` commands?
</blockquote></details>

Based on Richard's answer, I was confused why this table is nested under `information_schema`, since I read it as `information_schema` tables are what backs the `SHOW` commands`.

I think it's worth putting the `SHOW` table in its own top-level page (we have individual pages for each `SHOW` command but no over) or stick it under the top `Virtual schemas` page

**Review by ericharmeling — COMMENTED** [review]
TFTR @vy-ton and @richardjcai !

I think I've addressed all of your comments.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8909#-:-MW4twfkCgkksobbILjU:b-fsjeqo)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @jordanlewis, @knz, @RichardJCai, and @vy-ton)

___
*[_includes/sidebar-data-v20.2.json, line 2422 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw0WyQ-NxMrF7reNtj:-MW0Xz7w0BEziPC2lAqK:b-r1sn6l) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/_includes/sidebar-data-v20.2.json#L2422)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

2 suggestions:
* Put this closer to `Name Resolution` as it's the closest relative
* We should get Engineering input but I've been referring to this as `System Catalog`. Developers might not understand "virtual".
</blockquote></details>

I like both of those suggestions. 

1. Done

2. I updated the nav title, the title of the overview page ("Virtual Schemas" -> "System Catalogs"), the name of the file (`virtual-schemas.md`-> `system-catalogs.md`), and most of the instances of "virtual schema" to "system catalog" (where it made sense). 
___
*[v20.1/crdb-internal.md, line 15 at r3](https://reviewable.io/reviews/cockroachdb/docs/8909#-MT_sAF91S5nTsa_N9dl:-MW1-6iM9JTN4YCDLJZg:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/bb64665a60832d743e2963383b9e85d97e87b4b9/v20.1/crdb-internal.md#L15)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

to change without prior notice
</blockquote></details>

Done.
___
*[v20.2/information-schema.md, line 9 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw21atAW-wbphfdub2:-MW1-HV2DzTCfh5MPhed:br5mtbd) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v20.2/information-schema.md#L9)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

> can therefore be relied on to remain stable over time

We're not quite here yet with `information_schema`. I'm looking into this so will have more details soon
</blockquote></details>

I removed this entire paragraph.
___
*[v20.2/pg-catalog.md, line 7 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw2pCi1UJjnKKxDe-Q:-MW1-Jgz05UYGHjaC3jc:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v20.2/pg-catalog.md#L7)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

They correspond to both system catalog and system views, maybe use https://www.postgresql.org/docs/13/catalogs.html as the link
</blockquote></details>

Done.
___
*[v20.2/pg-catalog.md, line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw3CHW-MuD3kWq10Oq:-MW4ZeTq-a1_hKl_VRMG:b6k8ekq) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v20.2/pg-catalog.md#L13)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

Looking into the `None` tables [here](https://docs.google.com/document/d/1Y_S6-S_QfNewu70pil42XYU7rWLOwO0HAFiqtAL6Iao/edit#heading=h.dt3nanno7v0e), since I expected to be able to query empty `pg_catalog` tables
</blockquote></details>

See response below.
___
*[v20.2/pg-catalog.md, line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MTCmGO2whR4NoGStbVr-r5-13:-MW4Zg_r7lHBBtEDhaAC:b-b4g7o8) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v20.2/pg-catalog.md#L13)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

We should clarify what `None` here means. We have added tables like `pg_amop` as empty tables. So they are unimplemented but they do output in `SHOW TABLES` and you can query the empty table
</blockquote></details>

This is a list of the `pg_catalog` tables in the official 20.2.6 release. We should only document what is in the latest production release, for production-release docs (i.e., anything with the URL prefix https://www.cockroachlabs.com/docs/stable). 

I just built cockroach from the latest commit on the `release-20.2` branch, and see that all of the new, unimplemented tables are in `pg_catalog`. I'll follow up with an update after 20.2.7 is released (which is scheduled for March 29, but still has a few blockers).

The 21.1 beta (scheduled for Monday 3/22) should include all of the new, unimplemented `pg_catalog` tables. But IMO, because 21.1 is a testing release, the docs can include features that aren't in an "official release" in any case. 
___
*[v21.1/information-schema.md, line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw4ibcAWw2ctxh6u3i:-MW4sNQP7FXfp0Xr31nH:b-e99w4i) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v21.1/information-schema.md#L13)):*
> Is information_schema what powers SHOW commands?

`crdb_internal` is what powers `SHOW` commands.

> I was confused why this table is nested under information_schema...

Which table?

> I think it's worth putting the SHOW table in its own top-level page (we have individual pages for each SHOW command but no over) 

`SHOW TABLES` is a SQL statement, with its own page. Unless you are talking about an actual virtual table? I don't see a table called "`SHOW`". Or are you talking about `crdb_internal.create_statements`?

> or stick it under the top Virtual schemas page

I'd prefer to keep just the system catalog schemas at the top level of the Virtual Schemas page (now called the "System Catalogs" page)
___
*[v21.1/pg-catalog.md, line 13 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVw5Bxb21lr4un1wah4:-MW4j5lw1XvAxmQd9P-d:bsg5jhm) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v21.1/pg-catalog.md#L13)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

How are you determining the list for the first column? We might be falling into a trap of duplicating the list of Postgres pg_catalog. Instead of just documenting the ones we do implement.
</blockquote></details>

The list for the first column is exactly what is listed on this page (which includes all of the tables in `pg_catalog`, as well as a handful of system views): https://www.postgresql.org/docs/13/catalogs.html

The list for the second column is what is returned when I run `SHOW TABLES FROM pg_catalog` from a 3/17 build on the `release-21.1 branch.

The "None" values in the second columns indicate that a table/view in the postgres system catalog (https://www.postgresql.org/docs/13/catalogs.html) is missing from CRDB's `pg_catalog` schema in that 3/17 21.1 build.

The "(unimplemented)" indicates that a table is empty (I've changed "unimplemented"->"empty"). These are tables that are not in `pg_catalog` in 20.2.0-20.2.6, but are in `pg_catalog` in 21.1.beta-1 and 20.2.7. I assume that all of the tables recently added are empty. @richardjcai Please correct me if I am wrong.
___
*[v21.1/pg-catalog.md, line 17 at r5](https://reviewable.io/reviews/cockroachdb/docs/8909#-MVNxTJG6RpyVfFVTIK9-r5-17:-MW4pRkQ3hegdbacWbsz:b-7y7cvh) ([raw file](https://github.com/cockroachdb/docs/blob/ac8a611e71d1c84ff71795677a001fe152cbd1b7/v21.1/pg-catalog.md#L17)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

LGTM but piggybacking on Vy's comment, I think we should clarify that unimplemented means that the table exists but the rows are not populated.
</blockquote></details>

I changed "unimplemented" to "empty". "unimplemented" suggests that that will at one point be implemented. "empty" is a little clearer (i.e., they have no contents).

**Comment by RichardJCai** [comment]
> I changed "unimplemented" to "empty". "unimplemented" suggests that that will at one point be implemented. "empty" is a little clearer (i.e., they have no contents).

SGTM!

**Review by ianjevans — APPROVED** [review]

**Comment by ericharmeling** [comment]
TFTR, @ianjevans !

@vy-ton 

I'm going to go ahead and merge this, so we have the pages live. If we need to make any updates after it's been published, it should be pretty easy to do so.

**Pull request merged**
